src/sleep/sleep.c

```diff
@@ -287,7 +284,7 @@ static int execute(char **modes, char **states) {
 
         r = write_state(&f, states);
         if (r < 0)
-                return r;
+                return log_error_errno(errno, "Failed to write /sys/power/state: %m");
```

Comment:
write_state() returns the error as return value, hence the first argument to log_error_errno() should be `r`, not `errno`